File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

int score(coordinate_t landing_position) {
    float dist_to_center;
    dist_to_center = sqrt(pow(landing_position.x, 2) + pow(landing_position.y, 2));
    if (dist_to_center > 10.0F) {
        return 0;
    } else if (dist_to_center > 5.0F) {
        return 1;
    } else if (dist_to_center > 1.0F) {
        return 5;
    } else {
        return 10;
    }
}

File: solutions/r/darts/1/darts.R
score <- function(x, y) {
  radius2 = x ^ 2 + y ^ 2
  if (radius2 <= 1) return(10)
  if (radius2 <= 25) return(5)
  if (radius2 <= 100) return(1)
  else return(0)
}
